components/camel-weka/src/main/java/org/apache/camel/component/weka/WekaComponent.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.weka;
+
+import java.util.Map;
+
+import org.apache.camel.Endpoint;
+import org.apache.camel.component.weka.WekaConfiguration.Command;
+import org.apache.camel.spi.annotations.Component;
+import org.apache.camel.support.DefaultComponent;
+
+@Component("weka")
+public class WekaComponent extends DefaultComponent {
+
+    @Override
+    protected Endpoint createEndpoint(String urispec, String remaining, Map<String, Object> params) throws Exception {
+
+        // Init the configuration
+        WekaConfiguration config = new WekaConfiguration(this);
+        setProperties(config, params);
```

Comment:
I recall, you should do `setProperties` on config from the endpoint. Perhaps either @oscerd or @davsclaus can confirm this.

Reply:
Yes, you should this on the endpoint, so we could avoid reflections.

Reply:
Yes correct